crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -158,19 +178,36 @@ impl CoinGecko {
         token: &Token,
         denominator_price_eth: Decimal,
         prices: &HashMap<Token, f64>,
+        infos: &HashMap<Token, TokenInfo>,
```

Comment:
This method signature is quite 🤡 in my opinion (it has 5 values of which some are redundant).

I think we should simply pass it the token as `(TokenInfo, priceInEth)` and the denominator as `(Denominator, priceInEth)` and keep this method for the pure non-trivial math.

Reply:
The alternative to this function seems to be even more 🤡 error handling in the calling function. While your suggestion would make the conversion more pure it would make the surrounding code way harder to read.
Unless I'm missing some trick to reasonably manage errors (using anonymous closures to get early returns doesn't count IMO).

Reply:
I mean the fact that you are passing in `denominator_price_eth` on top of prices is just redundant (and seems inconsistent) so at least you could either
- remove denominator_price_eth and and "expect" on it being in prices 
- pass the token price into the function directly

But I also don't really see how error handling is an issue here (you are bubbling up errors in any case)